hashintel/hash
# Finalize `stateful` crate

This is part of [Better separation of crates](https://app.asana.com/0/1201707629991362/1201481007343142/f) (_internal_).

## 🌟 What is the purpose of this PR?

The `stateful` crate holds all kinds of different symbols for storing data. Internally it's aware of the Arrow-backend, which should be avoided to be exposed publicly.

## 🚫 Blocked by

- [ ] Add module level documentation
- [ ] Enhance crate-level documentation
- [ ] Improve doc-strings

## 🐾 Next steps

- Some arrow functionalities are exposed, those should be abstracted away
- The error library should be applied to this crate
- Conversion methods for `Agent` can probably be abstracted away

## 🔍 What does this change?

*TODO*

### 📜 Does this require a change to the docs?

The documentation was adjusted in this PR

## 🔗 Related links

- [Asana task](https://app.asana.com/0/0/1202092280081544/f) _(internal)_

## 🛡 What tests cover this?

The integration test suite covers this crate as well

## ❓ How to test this?

Tthis PR has very fine-grained commits and should be reviewed with this in mind. The big impact on line numbers and files mainly comes from moving symbols to another location. Commits, which are moving symbols, don't have any logic changes.

- Verify the changes listed in *What does this change?* under *Make components to be moved independent*
- Verify the new structure shown below

## 📹 Demo

*TODO*
